common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java

```diff
@@ -0,0 +1,231 @@
+package com.microsoft.identity.common.internal.request;
+
+import android.accounts.Account;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+import android.util.Pair;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.internal.authorities.Authority;
+import com.microsoft.identity.common.internal.broker.BrokerRequest;
+import com.microsoft.identity.common.internal.broker.BrokerValidator;
+import com.microsoft.identity.common.internal.logging.DiagnosticContext;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.oauth2.OpenIdConnectPromptParameter;
+import com.microsoft.identity.common.internal.ui.AuthorizationAgent;
+import com.microsoft.identity.common.internal.util.QueryParamsAdapter;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+public class MsalBrokerRequestAdapter implements IBrokerRequestAdapter {
+
+    private static final String TAG = MsalBrokerRequestAdapter.class.getName();
+
+    @Override
+    public BrokerRequest brokerRequestFromAcquireTokenParameters(@NonNull final AcquireTokenOperationParameters parameters) {
+
+        Logger.verbose(TAG, "Constructing result bundle from AcquireTokenOperationParameters.");
+
+        final BrokerRequest brokerRequest =  new BrokerRequest.Builder()
+                .authority(parameters.getAuthority().getAuthorityURL().toString())
```

Comment:
I assume AcquireTokenOperationParameters.mAuthority and Authority.getAuthorityURL() are always not null, right?

Reply:
That's right